### PR TITLE
[feature](nereids) Add subquery expression and In expression and TPCH…

### PR DESCRIPTION
**Pull request by zhengshiJ**
… sql

# Proposed changes

Issue Number: 

## Problem Summary:

1. Increase the expression of subquery and in.
2. Added tpch creation table and sql query, including original sql query and query rewritten by doris
3. Adjust the position of checkAnalyze
4. add exists subquery

## Checklist(Required)

1. Does it affect the original behavior: (No)
5. Has unit tests been added: (Yes)
6. Has document been added or modified: (No Need)
7. Does it need to update dependencies: (No)
8. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionType.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.trees.expressions;
+
+/**
+ * Types for all Expression in Nereids.
+ */
+public enum ExpressionType {
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionType.java:23]: #11066 already remove this type

Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/ExpressionType.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/In.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * In predicate expression.
+ */
+public class In extends Expression {
```
[the remark this answers is not in the document]
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/In.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java [hunk 3]
```diff
@@ -205,6 +208,18 @@ public R visitCaseWhen(CaseWhen caseWhen, C context) {
         return visit(caseWhen, context);
     }
 
+    public R visitInSubquery(InSubquery in, C context) {
+        return visit(in, context);
```
Comment (wangshuo128) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java:212]: visit SubqueryExpr by default, just like `visitEqualTo`.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java [hunk 4]
```diff
@@ -205,6 +208,18 @@ public R visitCaseWhen(CaseWhen caseWhen, C context) {
...
+    }
+
+    public R visitExistsSubquery(Exists exists, C context) {
```
Comment (wangshuo128) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java:215]: ditto.

File: fe/fe-core/src/test/java/org/apache/doris/nereids/tpch/AnalyzeTPCHTest.java [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.tpch;
+
+public class AnalyzeTPCHTest extends TPCHTestBase {
+    /*@Test
```
Comment (wangshuo128) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/tpch/AnalyzeTPCHTest.java:21]: Please add comments:
1. why comments?
2. when to remove these comments?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/SubqueryExpr.java [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+import com.google.common.base.Preconditions;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Subquery Expression.
+ */
+public class SubqueryExpr extends Expression {
+    protected LogicalPlan subquery;
```
Comment (wangshuo128) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/SubqueryExpr.java:35]: `subquery` -> `queryPlan`?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Exists.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        super(Objects.requireNonNull(subquery, "subquery can not be null"));
+    }
+
+    @Override
+    public DataType getDataType() throws UnboundException {
+        return BooleanType.INSTANCE;
+    }
+
+    @Override
+    public boolean nullable() throws UnboundException {
```
[the remark this answers is not in the document]
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Exists.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/SubqueryExpr.java [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+        // For multiple lines, struct type is returned, in the form of splicing,
+        // but it seems that struct type is not currently supported
+        return BooleanType.INSTANCE;
+    }
+
+    @Override
+    public boolean nullable() throws UnboundException {
+        // TODO:
+        // Any child is nullable, the whole is nullable
+        return false;
```
[the remark this answers is not in the document]
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/SubqueryExpr.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java (hunk 3 above)
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java (hunk 4 above)
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ExpressionVisitor.java:215]: done

File: fe/fe-core/src/test/java/org/apache/doris/nereids/tpch/AnalyzeTPCHTest.java (hunk 5 above)
Comment (zhengshiJ) [reply fe/fe-core/src/test/java/org/apache/doris/nereids/tpch/AnalyzeTPCHTest.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/SubqueryExpr.java (hunk 6 above)
Comment (zhengshiJ) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/SubqueryExpr.java]: done

**Review by 924060929 — APPROVED** [review]
LGTM

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**